internal/controller/serving/reconcilers/model_registry_reconciler.go

```diff
@@ -23,18 +28,31 @@ func NewModelRegistryInferenceServiceReconciler(client client.Client, log logr.L
 
 	err := client.List(context.Background(), &dscList)
 	if err != nil {
-		log.Error(err, "Failed to list DataScienceCluster")
+		return nil, fmt.Errorf("%w: %w", errGetDSC, err)
+	}
+
+	if len(dscList.Items) == 0 || len(dscList.Items) > 1 {
+		return nil, fmt.Errorf("%w: only one DataScienceCluster is allowed", errGetDSC)
+	}
+
+	dsc = dscList.Items[0]
+
+	ns, found, err := unstructured.NestedFieldCopy(dsc.Object, "spec", "components", "modelregistry", "registriesNamespace")
+	if err != nil || !found {
+		return nil, fmt.Errorf("%w: %w", errGetMRNamespaceFromDSC, err)
 	}
 
-	if len(dscList.Items) > 0 {
-		dsc = dscList.Items[0]
+	mrNamespaceFromDSC, isNsOk := ns.(string)
+	if !isNsOk {
+		return nil, fmt.Errorf("%w: invalid namespace", errGetMRNamespaceFromDSC)
+	}
 
-		ns, found, err := unstructured.NestedFieldCopy(dsc.Object, "spec", "components", "modelRegistry", "registriesNamespace")
-		if err == nil && found {
-			mrNamespaceFromDSC = ns.(string)
-		}
+	if mrNamespaceFromDSC == "" {
+		return nil, fmt.Errorf("%w: empty namespace", errGetMRNamespaceFromDSC)
 	}
 
+	log.Info("Model Registry Namespace from DataScienceCluster", "Namespace", mrNamespaceFromDSC)
```

Comment:
is the context of this message to log that MR was enabled?

Reply:
no if MR is not enabled the reconcile should not run as requested here https://github.com/opendatahub-io/odh-model-controller/pull/369#discussion_r1962871162 and worked in https://github.com/opendatahub-io/opendatahub-operator/pull/1682

this message is just to inform the user about which namespace is set to default if the inferenceservice does not specify a `modelregistry.opendatahub.io/namespace` label